### PR TITLE
V2 add feature branch to contributing 1750

### LINKED ISSUE #1750: Contributing.md: feature branch
### Overview
The [contributing guide ](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md)is an important documentation on situating new members to the team. Currently, it is lacking critical information about reviewing or creating non-default-branch (for example, the [wins-feature-1 branch](https://github.com/hackforla/website/tree/wins-feature-1)\).

### Action Items
- [x] Add feature branch information to the [contributing guide ](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md)
  - [x] As with above, place it at a place that is intuitive.
  - [x] Add a link to [GitHub's documentation on resolving merge conflicts
](https://docs.github.com/en/github/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github)

### Resources/Instructions
[Code base for wins-feature-1 branch](https://github.com/hackforla/website/tree/wins-feature-1)
[How to work off of a feature branch](https://github.com/hackforla/website/wiki/How-to-work-off-of-a-feature-branch)
[Resolving a merge conflict on GitHub](https://docs.github.com/en/github/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github)

### PR DESCRIPTION
**Pull request by plang-psm**
Fixes #1750

*** Please note this version will replace PR #3558.

### What changes did you make and why did you make them ?

3 changes were made to the Contributing page:
  - Add the "Working off of a feature branch" to wikis
  - Add the "Resolving a merge conflict on GitHub" to wikis.
  - This was not part of the action items but the table of contents link for 31.b.v was broken due to the spelling of How. I changed the 'H' to an 'h' which then fixed the link.

### CODE REVIEW

**Comment by arpitapandya** [comment]
Availability: 6 hrs
ETA: EOD 2022-09-20

**Comment by AzaniaBG** [comment]
ETA: 09/22/2022 by 5 PM
Availability: 09/21/2022: 3 PM - 5 PM; 09/22/2022: 3 PM - 5 PM

**Comment by arpitapandya** [comment]
@plang-psm There is a link missing to 2.8. When I click, gives error 404. I have other pointers which I would like to bring in next meeting.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -63,6 +63,10 @@ The following is a set of guidelines for contributing to the website repository,
         - [**ii. If there are conflicting changes in the upstream repository**](#ii-if-there-are-conflicting-changes-in-the-upstream-repository)
       - [**2.7.e Working on an issue (5): Incorporating changes from upstream**](#27e-working-on-an-issue-5-incorporating-changes-from-upstream)
         - [**i. Incorporating changes into your topic branch**](#i-incorporating-changes-into-your-topic-branch)
+    - [**2.8 Working off a feature branch**]()
```
Comment (AzaniaBG) [inline CONTRIBUTING.md]: Please add a link here.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -689,6 +693,54 @@ If you receive warnings about conflicts, abort the rebase with `git rebase --abo
...
+git pull name-of-remote-here feature-branch-name-here
+```
+
+#### **2.8.c Create a pull request**
+Stage and commit changes.
+
+```
+git add files-changed-here
+git commit -m 'Commit message'
+```
```
Comment (AzaniaBG) [inline CONTRIBUTING.md]: Please add instructions to push changes to the remote branch after line 734 (to match the other guide).

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -689,6 +693,54 @@ If you receive warnings about conflicts, abort the rebase with `git rebase --abo
...
+Create a pull request that is comparing across forks, from your branch to the feature branch with the following settings:
```
Comment (AzaniaBG) [inline CONTRIBUTING.md]: It might be helpful to specify that this is done on GitHub, rather than on the command line, but it's up to you :)

**Review by AzaniaBG — DISMISSED** [review]
@plang-psm ~ nice work on this issue. It was easy to understand all the changes you made. I left specific inline-comments for requested changes, summarized as follows:
* To match the instructions from [How to work off of a feature branch](https://github.com/hackforla/website/wiki/How-to-work-off-of-a-feature-branch), add the instruction to push changes after the instruction to stage and commit changes.
* The 2.8 link in the ToC needs a link
* You may want to add a link to the GitHub "resolving a merge conflict" in Section 2.8 so that people know where they can go if they run into that problem
* I tried the  3.1.b.v link on the [CONTRIBUTING.md](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#how-to-contribute-to-hack-for-la) and it is working. If it is working, it should not be changed, but maybe I'm not checking the correct link?

Let me know if you have any questions!

**Comment by arpitapandya** [comment]
Here are the notes that have been discussed in earlier meetings and last Thursday's office hours.
- The linked issue is written in 2021.
- At this time, Adding the same instructions to the `contributing.md` from the wiki page only will not help us.
- We have to update it based on the ongoing feedback received by the devs and the problems devs face while working on `feature-homepage-launch` (which includes working on issue, creating PR, running docker, and reviewing PR) and not having awareness of having `feature-homepage-launch` branch on their machine first while reviewing if they never picked up the small issue relating to `feature-homepage-launch` and worked on `gh-pages` only
- Also, we have to be more concerned about the order as it also reflects on other pages such as the pre-work checklist template.
- I and Kathryn were in process of opening an issue related to updating the `contributing guide` and making the instructions clear, especially for new devs who have no prior experience.
- Since we found out that we already have the issue and as Phillip already started working on it, I can't update the issue. I respect Phillip's time and contribution to this.
- To close this and approve it to be merged, We have found it more intuitive to just add a link to the wiki page under the resources section for now and not to create more confusion without clarity.
- I will create a new issue and help update the guide based on the feedback and dm's I have been receiving and past reviews.
- The New issue will focus on the instructions of the order of both branches and how to work on issues, create PR and review it, possible mistakes, and how to resolve them; and make devs aware of the feature branch and running docker.
- I welcome all the suggestions.

**Review by arpitapandya — APPROVED** [review]
@plang-psm Thank you for making all the changes and making yourself available to address the changes. Thank you @AzaniaBG for catching up on the errors and leaving valuable reviews which were really helpful. Both of your's contribution and time is much appreciated on this PR.
Great job!! @plang-psm @AzaniaBG

**Pull request merged**